### PR TITLE
feat: deprecated synths redemption

### PR DESCRIPTION
**Pull request by vbstreetz**

### CODE REVIEW

File: components/Currency/CurrencyAmount/CurrencyAmount.tsx [hunk 1]
```diff
@@ -25,11 +26,14 @@ export const CurrencyAmount: FC<CurrencyAmountProps> = ({
 	totalValue,
 	sign,
 	conversionRate,
+	showValue = true,
 	showTotalValue = true,
 	...rest
 }) => (
 	<Container {...rest}>
-		<Amount className="amount">{formatCurrency(amountCurrencyKey, amount)}</Amount>
+		{!showValue ? null : (
```
Comment (andytcf) [inline components/Currency/CurrencyAmount/CurrencyAmount.tsx:34]: ```showValue && <Amount className="amount">{formatCurrency(amountCurrencyKey, amount)}</Amount>``` will do the same job i think

File: components/Currency/CurrencyIcon/CurrencyIcon.tsx [hunk 2]
```diff
@@ -30,7 +32,23 @@ export const SNXIcon =
 export const getSynthIcon = (currencyKey: CurrencyKey) =>
 	`https://raw.githubusercontent.com/Synthetixio/synthetix-assets/master/synths/${currencyKey}.svg`;
 
-export const CurrencyIcon: FC<CurrencyIconProps> = ({ currencyKey, type, ...rest }) => {
+export const CurrencyIconContainer: FC<CurrencyIconProps> = (props) => (
```
Comment (andytcf) [inline components/Currency/CurrencyIcon/CurrencyIcon.tsx:35]: ```({isDeprecated})```

Comment (andytcf) [reply components/Currency/CurrencyIcon/CurrencyIcon.tsx:35]:
```
{isDeprecated && <DeprecatedXIconContainer>
				<Svg src={DeprecatedXIcon} />
			</DeprecatedXIconContainer>}
```

File: components/Currency/CurrencyName/CurrencyName.tsx [hunk 3]
```diff
@@ -20,17 +22,30 @@ export const CurrencyName: FC<CurrencyNameProps> = ({
...
+	const { t } = useTranslation();
+	return (
+		<Container {...{ showIcon }} {...rest}>
+			{showIcon && (
+				<CurrencyIcon className="icon" {...{ currencyKey, isDeprecated }} {...iconProps} />
+			)}
+			<NameAndSymbol>
+				<Symbol className="symbol">
+					{symbol || currencyKey}
+					{!isDeprecated ? null : (
```
Comment (andytcf) [inline components/Currency/CurrencyName/CurrencyName.tsx:38]:
```
{isDeprecated && <Deprecated>
							<DeprecatedDot></DeprecatedDot> {t('common.currency.deprecated')}
						</Deprecated>}
```

File: sections/synths/components/MobileAssetsTable.tsx [hunk 4]
```diff
@@ -106,13 +114,17 @@ const AssetsTable: FC<AssetsTableProps> = ({
...
+									{...{ showValue, showTotalValue }}
 								/>
 							</div>
 
 							<div>{currencyIsSynth ? asset.currencyKey : null}</div>
-							<div>
-								<SynthPriceCol currencyKey={asset.currencyKey as CurrencyKey} />
-							</div>
+
+							{!showPrice ? null : (
```
Comment (andytcf) [inline sections/synths/components/MobileAssetsTable.tsx:123]: ```{showPrice && <div><SynthPriceCol currencyKey={asset.currencyKey as CurrencyKey} /></div>}```

File: sections/synths/components/RedeemableDeprecatedSynthsModal.tsx [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+	const Redeemer = synthetixjs!.contracts.SynthRedeemer;
+
+	const { useContractTxn } = useSynthetixQueries();
+	const txn = useContractTxn(Redeemer, 'redeemAll', [redeemableDeprecatedSynthsAddresses], {
+		gasPrice: gasPrice.toBN(),
+	});
+
+	useEffect(() => {
+		switch (txn.txnStatus) {
+			// case 'prompting':
```
Comment (andytcf) [inline sections/synths/components/RedeemableDeprecatedSynthsModal.tsx:56]: probably better to handle the modal appearing here instead of handleRedeem

**Review by andytcf — CHANGES_REQUESTED** [review]
left some comments

File: components/Currency/CurrencyIcon/CurrencyIcon.tsx (hunk 2 above)
Comment (vbstreetz) [reply components/Currency/CurrencyIcon/CurrencyIcon.tsx:35]: the `props` are being based in the next line

File: components/Currency/CurrencyAmount/CurrencyAmount.tsx (hunk 1 above)
Comment (vbstreetz) [reply components/Currency/CurrencyAmount/CurrencyAmount.tsx]: fixed

File: components/Currency/CurrencyName/CurrencyName.tsx (hunk 3 above)
Comment (vbstreetz) [reply components/Currency/CurrencyName/CurrencyName.tsx:38]: fixed

File: sections/synths/components/MobileAssetsTable.tsx (hunk 4 above)
Comment (vbstreetz) [reply sections/synths/components/MobileAssetsTable.tsx]: fixed

File: sections/synths/components/RedeemableDeprecatedSynthsModal.tsx (hunk 5 above)
Comment (vbstreetz) [reply sections/synths/components/RedeemableDeprecatedSynthsModal.tsx]: fixed

**Pull request merged**